test/hotspot/jtreg/compiler/intrinsics/TestPopCountValueTransforms.java

```diff
@@ -43,17 +43,17 @@ public class TestPopCountValueTransforms {
 
     static final int SIZE = 4096;
 
-    static int rand_numI = G.uniformInts(0, Integer.MAX_VALUE).next();
-    static final int rand_bndI1 = G.uniformInts(0xF, Integer.MAX_VALUE).next();
-    static final int rand_bndI2 = G.uniformInts(0xFF, Integer.MAX_VALUE).next();
-    static final int rand_popcI1 = G.uniformInts(0, 2).next();
-    static final int rand_popcI2 = G.uniformInts(10, 20).next();
-
-    static long rand_numL = G.uniformLongs(0, Long.MAX_VALUE).next();
-    static final long rand_bndL1 = G.uniformLongs(0xFL, Long.MAX_VALUE).next();
-    static final long rand_bndL2 = G.uniformLongs(0xFFL, Long.MAX_VALUE).next();
-    static final long rand_popcL1 = G.uniformLongs(0, 3).next();
-    static final long rand_popcL2 = G.uniformLongs(20, 40).next();
+    static int rand_numI = G.uniformInts(Integer.MIN_VALUE, Integer.MAX_VALUE).next();
+    static final int rand_bndI1 = G.uniformInts(-0xFF, 0xFF).next();
+    static final int rand_bndI2 = G.uniformInts(-0xFFFFF, 0xFFFFF).next();
+    static final int rand_popcI1 = G.uniformInts(0, 4).next();
+    static final int rand_popcI2 = G.uniformInts(0, 32).next();
+
+    static long rand_numL = G.uniformLongs(Long.MIN_VALUE, Long.MAX_VALUE).next();
+    static final long rand_bndL1 = G.uniformLongs(-0xFFL, 0xFFL).next();
+    static final long rand_bndL2 = G.uniformLongs(-0xFFFFFFL, 0xFFFFFF).next();
+    static final long rand_popcL1 = G.uniformLongs(0, 4).next();
+    static final long rand_popcL2 = G.uniformLongs(0, 32).next();
```

Comment:
Can you please give us some code comments why you are doing:
- only uniform distribution. Is that needed? Generators generates special values more often for a good reason: it creates interesting edge cases, especially for bit operations like this here.
- Why are you restricting the ranges? There could always be surprises outside the ranges you pick, and it would be a shame to not generate those. Unless you are absolutely sure they are not needed. Or if extending the range would mean we would generate interesting cases with a probability that is too small, that could be another reason to restrict the ranges.

Reply:
@eme64 , addressed,